db/collection.go

```diff
@@ -93,16 +93,54 @@ func (c *Collection) Indexes() map[string]Index {
 // Adding an index will override any overlapping index values if they already exist.
 // @note: This does NOT currently build the index. If items have been added prior to adding
 // a new index, they will NOT be indexed a posteriori.
-func (c *Collection) AddIndex(path string, unique bool) error {
-	c.indexes[path] = Index{
+func (c *Collection) AddIndex(config IndexConfig) error {
+	indexKey := dsDBIndexes.ChildString(c.name)
+	exists, err := c.db.datastore.Has(indexKey)
+	if err != nil {
+		return err
+	}
+
+	indexes := map[string]IndexConfig{}
```

Comment:
Moved the storage format from `[]IndexConfig` to `map[string]IndexConfig` to make it easier to overwrite existing index configs and easier to detect an already existing index on `ID`

Reply:
Smart